include/s3select.h

```diff
@@ -2282,12 +2305,60 @@ class base_s3object
 	return m_returned_bytes_size;
   }
 
-  void result_values_to_string(multi_values& projections_resuls, std::string& result)
+  void json_result_format(multi_values& projections_results, std::string& result, std::string& output_delimiter, bool& col_ref)
   {
-    size_t i = 0;
+    std::string output_row_delimiter(1,m_csv_defintion.output_row_delimiter);
+    std::vector<std::string> key_path{};
+    std::vector<std::string> csv_header(m_sa->get_column_names());
+
+    for(auto& k: json_keys) {
```

Comment:
the outcome of this loop is always the same (json_keys never changed)
thus, this loop should not reside here on a very-intensive-loop.

Reply:
I think it is needed. This loop alter the keys obtained from the projection such that only the last part is needed.
eg. select _1.c1.c3.c4, _1.c3.c2
c4, c2 should obtained from this.

Reply:
I have moved the loop to set_base_definitions(). Is it ok?